ee/src/main/java/org/jboss/as/ee/subsystem/ManagedScheduledExecutorServiceAdd.java

```diff
@@ -65,7 +65,10 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
         final int coreThreads;
         final ModelNode coreThreadsModel = ManagedScheduledExecutorServiceResourceDefinition.CORE_THREADS_AD.resolveModelAttribute(context, model);
-        if (coreThreadsModel.isDefined()) {
+        if (coreThreadsModel.isDefined() && coreThreadsModel.asInt() == 0) {
+            //value 0 means unlimited number of threads
+            coreThreads = Integer.MAX_VALUE;
```

Comment:
This seems like a fairly dangerous change, despite the statement in the confluence docs[1].  Java thread pools are commonly configured and I don't think "0 means unlimited" is a common interpretation of the pool sizes; e.g. it is not the meaning of the pool size params in the JDK thread pool classes. A pool with a very large number of threads is very dangerous and IMO should require explicit configuration. (Note the confluence doc says the same "0 means no limit" thing about the core pool size  for the non-scheduled Managed Executor Services, and there I think such behavior would be wrong, as it conflicts with 'max-size'. It's also not implemented that way.)

Certainly a value of 0 is a poor choice for a ScheduledThreadPoolExecutor, as the class javadoc states, but if we want to disallow zero without rejecting existing configs that use it, perhaps we could just warn and treat 0 the same as undefined.

[1] https://docs.jboss.org/author/display/WFLY10/EE+Concurrency+Utilities+Configuration

Reply:
@bstansberry I also think, that it could be dangerous. I've fixed it according to doc.

 Better solution could be to let value 0 behave as if not defined at all -> it will set pool size to number of processor * 2. In that way it could be interpreted as not limited in the same way as not  defined (limited) at all...

What do you think?

Reply:
@bstansberry I'm sorry for my previous comment. I agree with you and will add warning and set behaviour 0 as undefined.